lib/optimizely/optimizely_user_context.rb

```diff
@@ -34,13 +34,15 @@ def initialize(optimizely_client, user_id, user_attributes)
       @user_id = user_id
       @user_attributes = user_attributes.nil? ? {} : user_attributes.clone
       @forced_decisions = {}
-      @qualified_segments = []
+      @qualified_segments = nil
+
+      @optimizely_client.odp_manager.identify_user(user_id: user_id)
```

Comment:
Can you check if this line handles a case where @optimizely_client is nil?
You might need a check if @ptimizely_client is not nil...
I haven't looked if such test exists in Rb, but I saw that in Py it does and it fails if the client is nil (test_get_variation__experiment_not_running).
Just a preemptive tip to check this.

Reply:
Good catch!